granulate_utils/metrics/spark.py

```diff
@@ -121,8 +121,15 @@ def _get_standalone_apps(self) -> Dict[str, Tuple[str, str]]:
         # https://github.com/apache/spark/blob/67a254c7ed8c5c3321e8bed06294bc2c9a2603de/core/src/main/scala/org/apache/spark/deploy/JsonProtocol.scala#L202
         metrics_json = rest_request_to_json(self._master_address, SPARK_MASTER_STATE_PATH)
         running_apps = {}
+        current_running_apps = []
 
-        for app in metrics_json.get("activeapps", []):
+        try:
+            current_running_apps = metrics_json["activeapps"]
+        except KeyError:
+            # Log the exception where the activeapps key is not found, and log the response
+            self._logger.exception("Could not find 'activeapps' in metrics_json", metrics_json=metrics_json)
```

Comment:
No need to log the exception backtrace itself (`logger.exception`) as we expect it. Just catch the KeyError and log a WARNING "No activeapps in metrics!"

Reply:
0a4f663